src/main/java/com/team2502/robot2017/subsystem/DriveTrainSubsystem.java

```diff
@@ -86,61 +86,39 @@ else if(xLevel < 0)
      * @param isLeftSide Whether or not it is the left joystick/side
      * @return the speed of the robot
      */
-    private double getSpeed(boolean isLeftSide)
+    private Pair<Double, Double> getSpeed()
     {
         double joystickLevel = 0.0D;
         // Get the base speed of the robot
-        joystickLevel = isLeftSide ? -OI.JOYSTICK_DRIVE_LEFT.getY() : -OI.JOYSTICK_DRIVE_RIGHT.getY();
+        joystickLevel = -OI.JOYSTICK_DRIVE_LEFT.getY();
 
         // Only increase the speed by a small amount
-        if(isLeftSide)
-        {
-            double diff = joystickLevel - lastLeft;
-            if(diff > 0.1D) { joystickLevel = lastLeft + 0.1D; }
-            else if(diff < 0.1D) { joystickLevel = lastLeft - 0.1D; }
-            lastLeft = joystickLevel;
-        }
-        else
-        {
-            double diff = joystickLevel - lastRight;
-            if(diff > 0.1D) { joystickLevel = lastRight + 0.1D; }
-            else if(diff < 0.1D) { joystickLevel = lastRight - 0.1D; }
-            lastRight = joystickLevel;
-        }
+        double diff = joystickLevel - lastLeft;
+        if(diff > 0.1D) { joystickLevel = lastLeft + 0.1D; }
+        else if(diff < 0.1D) { joystickLevel = lastLeft - 0.1D; }
+        lastLeft = joystickLevel;
 
-        // Sets the speed to 0 if the speed is less than 0.05 or larger than -0.05
-        if(Math.abs(joystickLevel) < 0.05D) { joystickLevel = 0.0D; }
+        Pair<Double, Double> out = new Pair<Double, Double>(joystickLevel, 0.0D);
 
+        joystickLevel = -OI.JOYSTICK_DRIVE_RIGHT.getY();
 
-        return joystickLevel;
-    }
+        diff = joystickLevel - lastRight;
+        if(diff > 0.1D) { joystickLevel = lastRight + 0.1D; }
+        else if(diff < 0.1D) { joystickLevel = lastRight - 0.1D; }
+        lastRight = joystickLevel;
 
-    private double getLeftSpeed()
-    {
-        return getSpeed(true);
-    }
+        // Sets the speed to 0 if the speed is less than 0.05 or larger than -0.05
+        if(Math.abs(joystickLevel) < 0.05D) { joystickLevel = 0.0D; }
 
-    private double getRightSpeed()
-    {
-        return getSpeed(false);
+        out.right = joystickLevel;
+
+        return out;
     }
 
     public void drive()
     {
-        double leftSpeed;
-        double rightSpeed;
-        if(DashboardData.getDriveType() == DriveTypes.DUAL_STICK)
-        {
-            leftSpeed = getLeftSpeed();
-            rightSpeed = getRightSpeed();
-        }
-        else
-        {
-            Pair<Double, Double> speed = getSpeedArcade();
-            leftSpeed = speed.getKey();
-            rightSpeed = speed.getValue();
-        }
-        drive.tankDrive(leftSpeed, rightSpeed, true);
+        Pair<Double, Double> speed = (DashboardData.getDriveType() == DriveTypes.DUAL_STICK) ? getSpeed() : getSpeedArcade();
+        drive.tankDrive(speed.left, speed.right, true);
```

Comment:
We have a joke at work about the "3:00am rule": Imagine it is 3:00 in the morning. There is a huge customer on the phone. You are skimming through the code trying to match anything you see that might, just might, explain the symptoms they are seeing. While this is more compact, you can't skim over it fast and still grok what it does. The old code's _style_ is way way way more 3:00am proof.

Actually, to be most accurate, it has to be 3:00am, AND you've been up for over a day straight AND there is no caffeine in sight... ;)